src/CSharp/CodeCracker/Refactoring/XmlDocumentationCodeFixProvider.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxKind;
+using System;
+using System.Collections.Generic;
+
+namespace CodeCracker.CSharp.Refactoring
+{
+    public abstract class XmlDocumentationCodeFixProvider : CodeFixProvider
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(DiagnosticId.XmlDocumentation.ToDiagnosticId());
+
+        public abstract SyntaxNode FixParameters(MethodDeclarationSyntax method, SyntaxNode root);
+
+        public abstract string MessageFormat { get; }
+
+        protected async Task<Document> FixParametersAsync(Document document, Diagnostic diagnostic, CancellationToken c)
+        {
+            var root = await document.GetSyntaxRootAsync(c).ConfigureAwait(false);
+            var documentationNode = (MethodDeclarationSyntax)root.FindNode(diagnostic.Location.SourceSpan);
```

Comment:
I caught an exception on this line:

````cshap
var documentationNode = (MethodDeclarationSyntax)root.FindNode(diagnostic.Location.SourceSpan);
````

Details:

````
Exception thrown: 'System.InvalidCastException' in CodeCracker.CSharp.dll

Additional information: Unable to cast object of type 'Microsoft.CodeAnalysis.CSharp.Syntax.PredefinedTypeSyntax' to type 'Microsoft.CodeAnalysis.CSharp.Syntax.MethodDeclarationSyntax'.
````

The problem is the node being found is the return type of the method. Even though the return type has a method **parent**, it is not the `MethodDeclaration` that is being found by the `FindNode` method. The most specific type is always found. Try to use `root.FindToken(diagnostic.Location.SourceSpan.Start).Parent.AncestorsAndSelf().OfType<MethodDeclarationSyntax>().First()` instead.